xdsl/transforms/loop_nest_lowering_utils.py

```diff
@@ -68,6 +62,105 @@ def indices_for_map(
     return output_indices
 
 
+def _insert_loop_nest(
+    rewriter: PatternRewriter,
+    insertion_point: InsertPoint,
+    zero_op: arith.Constant,
+    one_op: arith.Constant,
+    bounds: tuple[OpResult, ...],
+    iter_args: Sequence[SSAValue],
+    make_body: Callable[
+        [PatternRewriter, InsertPoint, Sequence[BlockArgument], Sequence[SSAValue]],
+        Sequence[SSAValue],
+    ],
+) -> Sequence[SSAValue]:
+    """
+    Creates a perfect loop nest, populating the innermost body with the provided
```

Comment:
SHould the title hint that we care about a perfect lops nest?

Reply:
what title?

Reply:
Sorry, I mean the function name

Reply:
Ah, maybe, although since it's a private function, I think it's sort of safe to drop the implicit perfection